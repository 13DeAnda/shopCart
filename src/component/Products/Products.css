.product{
    width: 233px;
    height: 209px;
    display: inline-block;
    margin-right: 50px;
}
.product img{
  width: auto;
  max-width: 100px;
  height: auto;
  max-height: 100px;
  padding: 5px;
}

.product .buttons{
  text-align: right;
  margin-top: -190px;
  float: right;
  display: inline-flex;
}

.glyphicon{
  color: green;
}
.glyphicon:hover{
  color: blue;
  cursor: pointer;
}

.addbutton{
  display: inline;
}

.product .glyphicon{
  margin-left:3px;
  margin-right: 3px;
}

.cartContainer{
  width: 309px;;
}

.offers{
  color: green;
}
.carousel-control{
  display: none;
}

.shopCartHeader{
  margin-bottom: 10px;
}

.tumbnail img{
  background: blue;
}
