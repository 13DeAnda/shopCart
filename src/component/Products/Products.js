import React from 'react';
import {Thumbnail, Button, Row, Col, Grid, Glyphicon, Collapse, Carousel} from 'react-bootstrap';

import ShopCart from '../ShopCart/ShopCart';

import './Products.css';

import _ from 'lodash';
export default class Products extends React.Component {

  constructor(props){
    super(props);

    this.addToCart = this.addToCart.bind(this);
    this.removeFromCart = this.removeFromCart.bind(this);
    this.displayCart = this.displayCart.bind(this);

    this.state = {
      cart: {},
      cartTotalItems: 0,
      showCart: true,
      products: {
        'CH1': {
          'code' : 'CH1',
          'name' : 'Chai',
          'price': 3.11,
          'image': '/assets/ch1.jpg',
          'quantity': 0
        },
        'AP1': {
          'code' : 'AP1',
          'name' : 'Apples',
          'price': 6.00,
          'image': '/assets/ap1.jpg',
          'quantity': 0
        },
        'CF1': {
          'code' : 'CF1',
          'name' : 'Coffe',
          'price': 11.23,
          'image': '/assets/cf1.jpg',
          'quantity': 0
        },
        'MK1': {
          'code' : 'MK1',
          'name' : 'Milk',
          'price': 4.75,
          'image': '/assets/mk1.jpg',
          'quantity': 0
        },
        'OM1': {
          'code' : 'OM1',
          'name' : 'Oatmeal',
          'price': 3.71,
          'image': '/assets/om1.jpg',
          'quantity': 0
        }
      }
    };
  }
  render() {
    var {products, cart, cartTotalItems, showCart} = this.state;
    return (
      <Grid className='Products'>
        <Row className='shopCartHeader '>
          <Col xs={8} className="text-center offers">
            <b> One year anniversary offers!</b>
            <Carousel>
              <Carousel.Item className='text-center'>
                 BOGO -- Buy-One-Get-One-Free Special on Coffee. (Unlimited)
              </Carousel.Item>
              <Carousel.Item className='text-center'>
                 APPL -- If you buy 3 or more bags of Apples, the price drops to $4.50.
              </Carousel.Item>
              <Carousel.Item className='text-center'>
                 CHMK -- Purchase a box of Chai and get milk free. (Limit 1)
              </Carousel.Item>
              <Carousel.Item className='text-center'>
                  APOM -- Purchase a bag of Oatmeal and get 50% off a bag of Apples
              </Carousel.Item>
            </Carousel>
          </Col>
          <Col xs={4} className="text-right">
            <Button bsSize="large" onClick={this.displayCart}>
              <Glyphicon glyph="shopping-cart" /> ({cartTotalItems})
            </Button>
          </Col>
        </Row>
        <Row>
          <Col sm={10}>
            {_.map(products, function(product, index){
              return(
                <Thumbnail className='product text-center' src={product.image}  key={index}>
                  <h4> $ {product.price}</h4>
                  <h6>{product.name} </h6> <br/>

                    <div className='buttons'>
                     {product.quantity > 0?
                          <div className='addbuttons'>
                            <Glyphicon glyph="minus-sign" onClick={()=> this.removeFromCart(product)} />

                            {product.quantity}
                          </div>
                        : '  '}
                        <div className='addbuttons'>
                          <Glyphicon glyph="plus-sign" onClick={() => this.addToCart(product)}/>
                        </div>
                    </div>
                </Thumbnail>
              );
            }.bind(this))}
          </Col>
          <Collapse in={showCart} mountOnEnter={true}>
            <div className="cartContainer">
              <ShopCart cart={cart}
                      addToCart={this.addToCart}
                      removeFromCart={this.removeFromCart} />
            </div>
          </Collapse>
        </Row>
      </Grid>
    );
  }

  //functions
  addToCart(product){
    console.log(this.state.cart['CF1']);
    var cart = _.cloneDeep(this.state.cart);
    var cartTotalItems = _.cloneDeep(this.state.cartTotalItems);

    product.quantity++;
    cartTotalItems++;
    cart[product.code] = product;

    this.setState({cart: cart, cartTotalItems: cartTotalItems});
    console.log("the cart", this.state.cart);
  }

  removeFromCart(product){
    var cart = _.cloneDeep(this.state.cart);
    var cartTotalItems = _.cloneDeep(this.state.cartTotalItems);

    product.quantity--;
    cartTotalItems--;
    cart[product.code] = product;

    if(product.quantity < 1){
      delete cart[product.code];
    }

    this.setState({cart: cart, cartTotalItems: cartTotalItems});
  }

  displayCart(){
    this.setState({showCart: !this.state.showCart});
  }
}
