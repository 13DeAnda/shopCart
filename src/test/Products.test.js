import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';


import Products from '../component/Products/Products';
import samples from './data/shopCart.json'

const productsWrapper = shallow(<Products />);
const productsInstance = productsWrapper.instance();


describe('Products component', () => {

  it('should render snapshot ShopCart', () => {
    const component = renderer.create(
      <Products />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('expects 5 elements displayed', function(){
    expect(productsWrapper.find('.product').length).toEqual(5);
  });

  it('clicks the first element expects the cart length be 1', function(){
    productsWrapper.find('.addProductButton').first().simulate('click');
    expect(Object.keys(productsWrapper.state().cart).length).toEqual(1);
  });

  it('expects total items to be 1', function(){
    expect(productsWrapper.state().cartTotalItems).toEqual(1);
  });

  it('clicks the first element expects the cart length be 1', function(){
    productsWrapper.find('.deleteProductButton').first().simulate('click');
    expect(Object.keys(productsWrapper.state().cart).length).toEqual(0);
  });
});
