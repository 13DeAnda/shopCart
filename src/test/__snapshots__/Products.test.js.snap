// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component should render snapshot ShopCart 1`] = `
<div
  className="Products container"
>
  <div
    className="shopCartHeader  row"
  >
    <div
      className="text-center offers col-xs-8"
    >
      <b>
         One year anniversary offers!
      </b>
      <div
        className="carousel slide"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <ol
          className="carousel-indicators"
        >
          <li
            className="active"
            onClick={[Function]}
          />
           
          <li
            className={null}
            onClick={[Function]}
          />
           
          <li
            className={null}
            onClick={[Function]}
          />
           
          <li
            className={null}
            onClick={[Function]}
          />
           
        </ol>
        <div
          className="carousel-inner"
        >
          <div
            className="text-center item active"
          >
            BOGO -- Buy-One-Get-One-Free Special on Coffee. (Unlimited)
          </div>
          <div
            className="text-center item"
          >
            APPL -- If you buy 3 or more bags of Apples, the price drops to $4.50.
          </div>
          <div
            className="text-center item"
          >
            CHMK -- Purchase a box of Chai and get milk free. (Limit 1)
          </div>
          <div
            className="text-center item"
          >
            APOM -- Purchase a bag of Oatmeal and get 50% off a bag of Apples
          </div>
        </div>
        <a
          className="carousel-control left"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            className="glyphicon glyphicon-chevron-left"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control right"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            className="glyphicon glyphicon-chevron-right"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
    <div
      className="text-right col-xs-4"
    >
      <button
        className="cartButton btn btn-lg btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="glyphicon glyphicon-shopping-cart"
        />
         (
        0
        )
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-10"
    >
      <div
        className="product text-center thumbnail"
      >
        <img
          alt={undefined}
          onError={undefined}
          onLoad={undefined}
          src="/assets/ch1.jpg"
        />
        <div
          className="caption"
        >
          <h4>
             $ 
            3.11
          </h4>
          <h6>
            Chai
             
          </h6>
           
          <br />
          <div
            className="buttons"
          >
              
            <div
              className="addbuttons"
            >
              <span
                className="addProductButton glyphicon glyphicon-plus-sign"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="product text-center thumbnail"
      >
        <img
          alt={undefined}
          onError={undefined}
          onLoad={undefined}
          src="/assets/ap1.jpg"
        />
        <div
          className="caption"
        >
          <h4>
             $ 
            6
          </h4>
          <h6>
            Apples
             
          </h6>
           
          <br />
          <div
            className="buttons"
          >
              
            <div
              className="addbuttons"
            >
              <span
                className="addProductButton glyphicon glyphicon-plus-sign"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="product text-center thumbnail"
      >
        <img
          alt={undefined}
          onError={undefined}
          onLoad={undefined}
          src="/assets/cf1.jpg"
        />
        <div
          className="caption"
        >
          <h4>
             $ 
            11.23
          </h4>
          <h6>
            Coffe
             
          </h6>
           
          <br />
          <div
            className="buttons"
          >
              
            <div
              className="addbuttons"
            >
              <span
                className="addProductButton glyphicon glyphicon-plus-sign"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="product text-center thumbnail"
      >
        <img
          alt={undefined}
          onError={undefined}
          onLoad={undefined}
          src="/assets/mk1.jpg"
        />
        <div
          className="caption"
        >
          <h4>
             $ 
            4.75
          </h4>
          <h6>
            Milk
             
          </h6>
           
          <br />
          <div
            className="buttons"
          >
              
            <div
              className="addbuttons"
            >
              <span
                className="addProductButton glyphicon glyphicon-plus-sign"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="product text-center thumbnail"
      >
        <img
          alt={undefined}
          onError={undefined}
          onLoad={undefined}
          src="/assets/om1.jpg"
        />
        <div
          className="caption"
        >
          <h4>
             $ 
            3.71
          </h4>
          <h6>
            Oatmeal
             
          </h6>
           
          <br />
          <div
            className="buttons"
          >
              
            <div
              className="addbuttons"
            >
              <span
                className="addProductButton glyphicon glyphicon-plus-sign"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-expanded={null}
      className="cartContainer collapse"
    >
      <div
        className="shopCart text-right"
      >
        <div
          className="row"
        >
          <div
            className="text-center"
          >
             
            <i>
              Cart is empty
            </i>
             
          </div>
        </div>
        <div
          className="text-right row"
        >
          <b>
            Total: 
          </b>
          $
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
